crates/bevy_asset/src/io/source.rs

```diff
@@ -0,0 +1,556 @@
+    pub fn with_processed_writer(
+        mut self,
+        writer: impl FnMut() -> Option<Box<dyn AssetWriter>> + Send + Sync + 'static,
+    ) -> Self {
+        self.processed_writer = Some(Box::new(writer));
+        self
+    }
+
+    /// Will use the given `watcher` function to construct processed [`AssetWatcher`] instances.
+    pub fn with_processed_watcher(
+        mut self,
+        watcher: impl FnMut(crossbeam_channel::Sender<AssetSourceEvent>) -> Option<Box<dyn AssetWatcher>>
+            + Send
+            + Sync
+            + 'static,
+    ) -> Self {
+        self.processed_watcher = Some(Box::new(watcher));
+        self
+    }
+
+    pub fn platform_default(path: &str, processed_path: &str) -> Self {
+        Self::default()
+            .with_reader(AssetSource::get_default_reader(path.to_string()))
+            .with_writer(AssetSource::get_default_writer(path.to_string()))
+            .with_watcher(AssetSource::get_default_watcher(
+                path.to_string(),
+                Duration::from_millis(300),
+            ))
+            .with_processed_reader(AssetSource::get_default_reader(processed_path.to_string()))
+            .with_processed_writer(AssetSource::get_default_writer(processed_path.to_string()))
+            .with_processed_watcher(AssetSource::get_default_watcher(
+                processed_path.to_string(),
+                Duration::from_millis(300),
+            ))
+    }
+}
+
+/// A [`Resource`] that hold (repeatable) functions capable of producing new [`AssetReader`] and [`AssetWriter`] instances
+/// for a given asset source.
+#[derive(Resource, Default)]
+pub struct AssetSourceBuilders {
+    sources: HashMap<CowArc<'static, str>, AssetSourceBuilder>,
+    default: Option<AssetSourceBuilder>,
+}
+
+impl AssetSourceBuilders {
+    /// Inserts a new builder with the given `id`
+    pub fn insert(&mut self, id: impl Into<AssetSourceId<'static>>, source: AssetSourceBuilder) {
+        match id.into() {
+            AssetSourceId::Default => {
+                self.default = Some(source);
+            }
+            AssetSourceId::Name(name) => {
+                self.sources.insert(name, source);
+            }
+        }
+    }
+
+    /// Gets a mutable builder with the given `id`, if it exists.
+    pub fn get_mut<'a, 'b>(
+        &'a mut self,
+        id: impl Into<AssetSourceId<'b>>,
+    ) -> Option<&'a mut AssetSourceBuilder> {
+        match id.into() {
+            AssetSourceId::Default => self.default.as_mut(),
+            AssetSourceId::Name(name) => self.sources.get_mut(&name.into_owned()),
+        }
+    }
+
+    /// Builds an new [`AssetSources`] collection. If `watch` is true, the unprocessed sources will watch for changes.
+    /// If `watch_processed` is true, the processed sources will watch for changes.
+    pub fn build_sources(&mut self, watch: bool, watch_processed: bool) -> AssetSources {
```

Comment:
There is currently no case for it. However this can't be collapsed into a single bool as there are 3 states to capture (both off, watch unprocessed, and watch processed).

Reply:
Should probably be an enum then. This is just a nit.